pkg/chartutil/jsonschema.go

```diff
@@ -91,3 +99,51 @@ func ValidateAgainstSingleSchema(values Values, schemaJSON []byte) (reterr error
 
 	return nil
 }
+
+// keep the old behaviour for empty $schema or the one that defined in
+// https://github.com/xeipuuv/gojsonschema/blob/v1.2.0/draft.go#L46-L62
+func validateUsingNewValidator(valuesJSON, schemaJSON []byte) (bool, error) {
```

Comment:
This function does four things:
1. determines the jsonschema version via partial JSON load (exiting early if no `$schema` key present)
2. Determine/filter the jsonschema version to use 
3. Load the chart's schema and values as JSON
4. Validates the values against the schema

It might be worth factoring out some of these logical functionalities into separate functions?

Reply:
I don't see the necessity to split up this function. IMO, factoring out is only worth it if:
- reusability, in this case, no other code reuse it, we can split it later if in the future turns out that some of the code can be used by other code
- readability, IMO, in this case, splitting up the function harms readability, you need to jump between functions to understand it, it is only worth it if we have complex branching in the function, in this case, we don't have branching

Reply:
Another good reason would be to make the code more testable, but that doesn't seem to be the case here.

My take on @gjenkins8 comment is this is doing more than validating

Reply:
> My take on @gjenkins8 comment is this is doing more than validating

I agree with this, but looking at the function `ValidateAgainstSingleSchema` before this PR, it does everything in a single function.

At first, I even put all the code inside that `ValidateAgainstSingleSchema` function, but then I tried my best not to touch existing code as much as possible, by separating this function, I only added 3 line to the original function

Reply:
also, the function is only around 50 lines, not sure how to break it.